core/src/main/java/org/frankframework/core/Adapter.java

```diff
@@ -194,32 +193,26 @@ public void configure() throws ConfigurationException { //TODO check if we shoul
 			configureReceiver(receiver);
 		}
 
-		List<String> hrs = new ArrayList<>();
-		for (IPipe pipe : pipeline.getPipes()) {
-			if (pipe instanceof AbstractPipe aPipe && StringUtils.isNotEmpty(aPipe.getHideRegex()) && !hrs.contains(aPipe.getHideRegex())) {
-				hrs.add(aPipe.getHideRegex());
-			}
-		}
-		StringBuilder sb = new StringBuilder();
-		for (String hr : hrs) {
-			if (!sb.isEmpty()) {
-				sb.append("|");
-			}
-			sb.append("(");
-			sb.append(hr);
-			sb.append(")");
-		}
-		if (!sb.isEmpty()) {
-			composedHideRegex = sb.toString();
-		}
-
+		composedHideRegex = computeCombinedHideRegex();
 		if(runState.getRunState()==RunState.ERROR) { // if the adapter was previously in state ERROR, after a successful configure, reset it's state
 			runState.setRunState(RunState.STOPPED);
 		}
 
 		configurationSucceeded = true; //Only if there are no errors mark the adapter as `configurationSucceeded`!
 	}
 
+	@Nonnull
+	String computeCombinedHideRegex() {
+		String combinedHideRegex = pipeline.getPipes().stream()
+				.map(IPipe::getHideRegex)
+				.filter(StringUtils::isNotEmpty)
+				.distinct()
+				.collect(Collectors.joining(")|(", "(", ")"));
+
+		if ("()".equals(combinedHideRegex)) return "";
```

Comment:
Kwestie van stijl denk ik - maar ik zou dit voor de leesbaarheid uitschrijven:

```suggestion
		if ("()".equals(combinedHideRegex)) {
			return "";
		}
```